homework/list.js

```diff
@@ -1,79 +1,100 @@
-'use strict';
+﻿'use strict';
 
-(function() {
+(function (exports) {
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
+    var TodoListManager = function () {
+        this._listNoteContent = [];
+        this._wrapper = null;
+    }
 
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
-    window.addEventListener('click', function(event) {
-      onNoteOpen(event);
-    });
-  }
+    TodoListManager.prototype = {
 
-  function onNoteOpen(event) {
-    if (event.target.classList.contains('note-title')) {
-      var id = event.target.dataset.noteId;
-      var content = _listNoteContent[id];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    };
-  }
+        start() {
+            this._wrapper = document.querySelector('#note-list-wrapper');
 
-  function preloadFirstNote() {
-    if (_listNoteContent.length !== 0) {
-      var content = _listNoteContent[0];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    }
-  }
+            this.fetchList()
+         .then((function (data) {
+             this.updateList(data);
+             this.drawList();
+             this.preloadFirstNote();
+         }).bind(this));
+            window.addEventListener('click', (function (event) {
+                this.onNoteOpen(event);
+            }).bind(this));
+        },
+
+        onNoteOpen(event) {
+            if (event.target.classList.contains('note-title')) {
+                var id = event.target.dataset.noteId;
+                var content = _listNoteContent[id];
+                window.dispatchEvent(new CustomEvent('note-open',
+                  { detail: content }));
+            };
+        },
+        handleEvent(event) {
```

Comment:
In line 21 you `addEventListener` of click to an anonymous function, rather than to assign the `this` as the listener. So this method will do nothing and never be called.